### PR TITLE
First Funky(Tm) attachment

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request
HEY YOU! WOULDNT IT BE COOL IF YOU HAD AN ITEM THAT COULD MAKE ANY GUN YOU WANT BE LEVER ACTION??!!
WELL THEN THIS IS FOR YOU!

Introducing the new "lever-action conversion kit".
Found as research loot, this attachment will halve the fire delay of any gun in exchange for making it lever action.
You want a GL with twice the firing speed? The cost will be carpal tunnel, but it will be worth it.

I have also divided the attachment file into a folder so each attachment type gets their own file.

## Why It's Good For The Game
The new gun system will allow for a lot of wacky things. This will add a minor wacky attachment to research content.
I dont see this having any real balance consequences.

## Changelog
:cl:
add: New gun attachment: Lever-action conversion kit. Halves any guns fire rate in exchange for it being made lever action. This is found as research loot.
/:cl:

### CODE REVIEW

**Comment by SilencedMP5A5** [comment]
time for a lever action revolver :)

**Comment by Three333Digit** [comment]
basically https://www.youtube.com/watch?v=6oMUiZXYQC4

**Comment by MereWalker** [comment]
Lever action minigun... LETS GOOOOOOO

**Comment by novaepee** [comment]
I like the idea of that the more marines robust xenomorphs the more toys marines get.

**Comment by Pariah919** [comment]
gonna make a macro

**Comment by ThelastArchiver** [comment]
cant wait for autosniper to shred through boilers via now ( even more) insane damage

**Review by BraveMole — CHANGES_REQUESTED** [review]
Ok so: 

- The PR is unreviewable, 1800 lines change for one attachment. I could maybe navigate if you made some commits to help, but right there, i can't see what you did.
- I asked you to make one attachment to prove that your concept is balancable, you added an attachment that halves the fire rate to all the guns; if this is not balance hell, i don't know what is.
- Even as a concept, smashing a key ==> better firerate is not something i'd like to promote.

**Comment by Mechanelize** [comment]
unbased maintainer vs chad 1800 line coder

**Comment by Pill-U** [comment]
did you based mf refactor attachment code for ONE meme attachment or what the fuck am I looking at

File: code/modules/projectiles/attachments/underbarrel.dm [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+			to_chat(user, span_notice("You are now using [src]."))
+		is_active = TRUE
+		overlays += image('icons/Marine/marine-weapons.dmi', src, "active")
+		. = TRUE
+	for(var/X in master_gun.actions)
+		var/datum/action/A = X
+		A.update_button_icon()
+	update_icon()
+
+/obj/item/attachable/lever_action
```
Comment (XSlayer300) [inline code/modules/projectiles/attachments/underbarrel.dm:228]: The lever action shenanigans start here by the way.

File: code/modules/projectiles/attachments/underbarrel.dm [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	pixel_shift_y = 18
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_BYPASS_ALLOWED_LIST //Fuckit. I want to see this on anything.
+
+/obj/item/attachable/lever_action/can_attach(obj/item/attaching_to, mob/attacher)
+	. = ..()
+	if(!. || !isgun(attaching_to))
+		return
+	var/obj/item/weapon/gun/gun = attaching_to
+	if(CHECK_BITFIELD(gun.reciever_flags, AMMO_RECIEVER_REQUIRES_UNIQUE_ACTION))
+		to_chat(attacher, span_warning("You cannot attach [src] to a gun that is already manual."))
```
Comment (XSlayer300) [inline code/modules/projectiles/attachments/underbarrel.dm:243]:
```suggestion
		to_chat(attacher, span_warning("You cannot attach [src] to a gun that is already manually operated."))
```

**Comment by Tristan63** [comment]
image0

**Comment by BraveMole** [comment]
Closing as stale

**Pull request closed without merging**